docs/src/simulation_tips.md

```diff
@@ -191,11 +192,66 @@ to achieve this
 ### Arrays in GPUs are usually different from arrays in CPUs
 
 On the CPU Oceananigans.jl uses regular `Array`s, but on the GPU it has to use `CuArray`s
-from the CUDA.jl package. You might need to keep this difference in mind when using arrays
-to `set!` initial conditions or when using arrays to provide boundary conditions and
-forcing functions.
+from the CUDA.jl package. While deep down both are arrays, their implementations are different
+and both can behave very differently. For example if can be difficult just view a `CuArray`.
+Consider the example below:
+
+```julia
+julia> using Oceananigans; using Adapt
+
+julia> grid = RegularRectilinearGrid(size=(1,1,1), extent=(1,1,1))
+RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}
+                   domain: x ∈ [0.0, 1.0], y ∈ [0.0, 1.0], z ∈ [-1.0, 0.0]
+                 topology: (Periodic, Periodic, Bounded)
+  resolution (Nx, Ny, Nz): (1, 1, 1)
+   halo size (Hx, Hy, Hz): (1, 1, 1)
+grid spacing (Δx, Δy, Δz): (1.0, 1.0, 1.0)
+
+julia> model = IncompressibleModel(grid=grid, architecture=GPU())
+IncompressibleModel{GPU, Float64}(time = 0 seconds, iteration = 0) 
+├── grid: RegularRectilinearGrid{Float64, Periodic, Periodic, Bounded}(Nx=1, Ny=1, Nz=1)
+├── tracers: (:T, :S)
+├── closure: IsotropicDiffusivity{Float64,NamedTuple{(:T, :S),Tuple{Float64,Float64}}}
+├── buoyancy: SeawaterBuoyancy{Float64,LinearEquationOfState{Float64},Nothing,Nothing}
+└── coriolis: Nothing
+
+julia> typeof(model.velocities.u.data)
+OffsetArrays.OffsetArray{Float64,3,CUDA.CuArray{Float64,3}}
+
+julia> adapt(Array, model.velocities.u.data)
+3×3×3 OffsetArray(::Array{Float64,3}, 0:2, 0:2, 0:2) with eltype Float64 with indices 0:2×0:2×0:2:
+[:, :, 0] =
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+[:, :, 1] =
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+
+[:, :, 2] =
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+ 0.0  0.0  0.0
+```
+
+Notice that in order to view the `CuArray` that stores values for `u` we needed to transform
+it into a regular `Array` first using `Adapt.adapt`. If we naively try to view the `CuArray`
+without that step we get an error:
 
-To learn more about working with `CuArray`s, see the
+```julia
+julia> model.velocities.u.data
+3×3×3 OffsetArray(::CUDA.CuArray{Float64,3}, 0:2, 0:2, 0:2) with eltype Float64 with indices 0:2×0:2×0:2:
+[:, :, 0] =
+Error showing value of type OffsetArrays.OffsetArray{Float64,3,CUDA.CuArray{Float64,3}}:
+ERROR: scalar getindex is disallowed
```

Comment:
This error can also be alleviated by writing `CUDA.allowscalar(true)`. Scalar operations are legal, but _disabled_ by default because they are slow. Sometimes its nice to enable scalar operations when working at the REPL and prototyping.

Reply:
Included a remark!

Reply:
Although I fear this passage is about to become obsolete very soon. Testing these with CUDA.jl 2.6 in Julia 1.6 works out of the box. Even if I explicitly ask CUDA to not allow scalar operations:

```julia
julia> CUDA.allowscalar(false)

julia> A = rand(1:10, (5,5,5));

julia> d_A = CuArray(A);

julia> d_A
5×5×5 CuArray{Int64, 3}:
[:, :, 1] =
  5  4  1  5   5
  9  1  1  1   5
 10  4  6  7  10
 10  5  7  2   3
  8  5  1  8   6

...
```

Reply:
Yeah, I think it makes a ton of sense to enable scalar operations for printing purposes...

Reply:
I think the error users get is something like "Could not compile <kernel with long unintelliglbe name> because on of the arguments is not isbits" or something like that. Should we help users understand that this error often occurs because they have tried to use an `Array` on the GPU when they should have used `CuArray`?

Reply:
Up to you, the changes so far are an improvement as far as I am concerned. This is also not a very common error, because functions are a better, faster choice for implementing boundary conditions and forcing functions. The few users that need to use arrays (mostly for boundary conditions) are doing something sophisticated, like coupling oceananigans to an external model.